components/layout/display_list_builder.rs

```diff
@@ -1683,26 +1681,22 @@ impl BlockFlowDisplayListBuilding for BlockFlow {
         } else {
             self.layer_id(0)
         };
-        let paint_layer = PaintLayer::new(layer_id, color::transparent(), scroll_policy);
         let stacking_context = self.fragment.create_stacking_context(
             &self.base,
             display_list,
             layout_context,
-            StackingContextLayer::Existing(paint_layer),
+            StackingContextLayerNecessity::Always(layer_id, scroll_policy),
             stacking_context_creation_mode);
 
         let outermost_stacking_context = match outer_display_list_for_overflow_scroll {
             Some(mut outer_display_list_for_overflow_scroll) => {
                 outer_display_list_for_overflow_scroll.children.push_back(stacking_context);
 
-                let paint_layer = PaintLayer::new(self.layer_id(0),
-                                                  color::transparent(),
-                                                  scroll_policy);
                 self.fragment.create_stacking_context(
                     &self.base,
                     outer_display_list_for_overflow_scroll,
                     layout_context,
-                    StackingContextLayer::Existing(paint_layer),
+                    StackingContextLayerNecessity::Always(layer_id, scroll_policy),
```

Comment:
I inadvertently changed this LayerId from ```self.layer_id(0)``` to ```layer_id```. I will fix this and upload a new version of the patch. This should fix the test failure.